src/core/public/chrome/recently_accessed/recently_accessed_service.test.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { httpServiceMock } from '../../http/http_service.mock';
+import { RecentlyAccessedService } from './recently_accessed_service';
+import { Subject } from 'rxjs';
+import { takeUntil, bufferCount } from 'rxjs/operators';
+
+// Maybe this should be moved to our global jest polyfills?
+class LocalStorageMock implements Storage {
+  private store = new Map<string, string>();
+
+  clear() {
+    this.store = new Map<string, string>();
+  }
+
+  getItem(key: string) {
+    return this.store.get(key) || null;
+  }
+
+  setItem(key: string, value: any) {
+    this.store.set(key, value.toString());
+  }
+
+  removeItem(key: string) {
+    this.store.delete(key);
+  }
+
+  key(index: number) {
+    return [...this.store.keys()][index] || null;
+  }
+
+  get length() {
+    return this.store.size;
+  }
+}
+
+describe('RecentlyAccessed#start()', () => {
+  // @ts-ignore
+  beforeAll(() => (global.localStorage = new LocalStorageMock()));
+  beforeEach(() => localStorage.clear());
+  // @ts-ignore
+  afterAll(() => (global.localStorage = undefined));
```

Comment:
Can we cache the previous value of `localStorage` and restore it?

```js
const cachedStorage = global.localStorage;
// ...
afterAll(() => (global.localStorage = cachedStorage));
```

Reply:
Sure, there is no value right now, but good to do anyways.